### DISCUSSION TITLE
How to use screen.on?

### DISCUSSION THREAD

**Discussion by Rysakov1986** [nut-tree/nut.js#503] [General]
Please help me. 
How to use screen.on?
Can you give me some examples?

Thanks!

**Comment by Rysakov1986** [comment]
I add hook:
`screen.on(await imageResource("test.jpg"), (window) => { console.log('test');})`
but, what is the next step?

**Reply by s1hofmann** [reply]
@Rysakov1986 First you register a [MatchResultCallback](https://nut-tree.github.io/apidoc/types/screen_class.MatchResultCallback.html) for a certain `find` resource. This can be an `imageResource` or a different [type of `query`](https://nut-tree.github.io/apidoc/modules/query_class.html)

```
const img = await imageResource("test.jpg");
screen.on(img , (result: MatchResult) => {
  console.log('test');
});
```

Once a hook is registered it will be called whenever the reference you registered it on is found via one of the on-screen search functions like [`find`](https://nut-tree.github.io/apidoc/classes/screen_class.ScreenClass.html#find), [`findAll`](https://nut-tree.github.io/apidoc/classes/screen_class.ScreenClass.html#findAll) or [`waitFor`](https://nut-tree.github.io/apidoc/classes/screen_class.ScreenClass.html#waitFor)
